src/main/java/com/alevelhw/module2/service/AnalyticsService.java

```diff
@@ -0,0 +1,116 @@
+    }
+
+    public static void printNumbersSoldTelevisions(Map<String, Invoice> invoiceMap) {
+        Set<Map.Entry<String, Invoice>> invoiceSet = invoiceMap.entrySet();
+        int numbersSoldTelevisions = (int) invoiceSet.stream()
+                .map(x -> x.getValue()
+                        .getTelevisionsSet()
+                        .size())
+                .mapToInt(x -> x)
+                .summaryStatistics()
+                .getSum();
+        System.out.println("Количество проданных телевизоров: " + numbersSoldTelevisions);
+    }
+
+    public static void printLeastInvoice(Map<String, Invoice> invoiceMap, ShopService shopService) {
+        Set<Map.Entry<String, Invoice>> invoiceSet = invoiceMap.entrySet();
+        Comparator<Invoice> comparator = Comparator.comparingInt(shopService::getTotalPriceInvoice);
+        Invoice leastInvoice = invoiceSet.stream()
+                .map(Map.Entry::getValue)
+                .min(comparator)
+                .orElse(new Invoice());
+        System.out.printf("Сумма наименьшего чека у покупателя: %s и составляет: %s\n",
+                leastInvoice.getCustomer(), shopService.getTotalPriceInvoice(leastInvoice));
+    }
+
+    public static void printSumOfOllPurchases(Map<String, Invoice> invoiceMap, ShopService shopService) {
+        Set<Map.Entry<String, Invoice>> invoiceSet = invoiceMap.entrySet();
+        int sumOllPurchase = invoiceSet.stream()
+                .map(Map.Entry::getValue)
+                .map(shopService::getTotalPriceInvoice)
+                .mapToInt(x -> x)
+                .sum();
+        System.out.println("Сумма всех покупок: " + sumOllPurchase);
+    }
+
+    public static void printQuantityRetails(Map<String, Invoice> invoiceMap) {
+        Set<Map.Entry<String, Invoice>> invoiceSet = invoiceMap.entrySet();
+        int quantityRetail = (int) invoiceSet.stream()
+                .map(Map.Entry::getValue)
+                .filter(x -> x.getType().equals("retail"))
+                .count();
+        System.out.println("Количество инвойсов категории retail: " + quantityRetail);
+    }
+
+    public static void printInvoicesWithOneTypeProducts(Map<String, Invoice> invoiceMap) {
+        Set<Map.Entry<String, Invoice>> invoiceSet = invoiceMap.entrySet();
+        int invoicesWithOneTypeProducts = (int) invoiceSet.stream()
+                .map(Map.Entry::getValue)
+                .filter(x -> x.getTelephonesSet().size() == 0 || x.getTelevisionsSet().size() == 0)
+                .count();
+        System.out.println("Количество инвойсов которые содержат только один тип товара: "
+                + invoicesWithOneTypeProducts);
+    }
+
+    public static void printFirstThreeInvoices(Map<String, Invoice> invoiceMap) {
+        Set<Map.Entry<String, Invoice>> invoiceSet = invoiceMap.entrySet();
+        System.out.println("Первые три инвойса:");
+        invoiceSet.stream()
+                .limit(3)
+                .forEach(System.out::println);
```

Comment:
It is better to add time to the order and sort by time, before specifying the limit